wikimedia/wikipedia-ios
# [YiR] Learn more web view

**Phabricator:** https://phabricator.wikimedia.org/T378839

### Notes
This PR allows for a "Donate" button overlay to be displayed on top of SinglePageWebViewController, similar to how we display the "Return" button on the thank you web view.

I refactored SinglePageWebViewController a bit to allow the passing in of a `configType` enum, along with associated config data. I'm not sure if this clears it up much but I was getting a bit mixed up in that file without the cleanup.

I also added a developer settings toggle for "Bypass donation". This causes a "fake success" in both the native and web donate flows, so that we can test successful donation navigation without having to make an actual donation.

This PR depends on the work in https://github.com/wikimedia/wikipedia-ios/pull/5048. Please merge that PR first.

### Test Steps
Note: This can be tested more easily once the donate slide UI work is in.

1. Ensure simulator region and primary app language lines up with YiR target countries / languages. Log into account with edits.
2. Go to developer settings menu, toggle both YiR feature flag and "Bypass donation". Background, foreground the app. YiR entry point should appear.
3. Temporarily add test code from https://github.com/wikimedia/wikipedia-ios/commit/ac59c77d223444ace30ac9336ac9375389aa59d8,.
4. Go through YiR flow. On last slide, tap "Done". The temp code mimics how this will act once the "learn more" button is tapped on the Donate slide.
5. Ensure web view appears with "Donate" button on top. Tap Donate button. Ensure donate flow appears.
6. Donate from one of the forms. You should be able to use fake wallet details in simulator now. Developer settings toggle should allow donation to suceeed.
7. Ensure you are popped back to learn more web view after donation.
